From b4fd6a8064819e090d3edea0ef50cbd5997e6cef Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Tue, 7 Apr 2015 18:02:14 +0200
Subject: [PATCH] Access oVirt API through /ovirt-engine/api rather than /api

/api is a deprecated entry point.

(cherry picked from commit 3c048fa41a4532e77c6e5c5489404202581f99e5)

Resolves: rhbz#1277447
---
 govirt/ovirt-proxy-deprecated.c | 4 ++--
 govirt/ovirt-proxy.c            | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/govirt/ovirt-proxy-deprecated.c b/govirt/ovirt-proxy-deprecated.c
index fe06703..1653baa 100644
--- a/govirt/ovirt-proxy-deprecated.c
+++ b/govirt/ovirt-proxy-deprecated.c
@@ -61,7 +61,7 @@ gboolean ovirt_proxy_fetch_vms(OvirtProxy *proxy, GError **error)
 
     g_return_val_if_fail(OVIRT_IS_PROXY(proxy), FALSE);
 
-    vms_node = ovirt_proxy_get_collection_xml(proxy, "/api/vms", error);
+    vms_node = ovirt_proxy_get_collection_xml(proxy, "/ovirt-engine/api/vms", error);
     if (vms_node == NULL)
         return FALSE;
 
@@ -104,7 +104,7 @@ void ovirt_proxy_fetch_vms_async(OvirtProxy *proxy,
                                          * would trigger a deprecation
                                          * warning */
                                         fetch_vms_async_cb);
-    ovirt_proxy_get_collection_xml_async(proxy, "/api/vms", result, cancellable,
+    ovirt_proxy_get_collection_xml_async(proxy, "/ovirt-engine/api/vms", result, cancellable,
                                          fetch_vms_async_cb, NULL, NULL);
 }
 
diff --git a/govirt/ovirt-proxy.c b/govirt/ovirt-proxy.c
index f1b88cb..a5877ff 100644
--- a/govirt/ovirt-proxy.c
+++ b/govirt/ovirt-proxy.c
@@ -712,7 +712,7 @@ static void ovirt_proxy_set_session_id(OvirtProxy *proxy, const char *session_id
     proxy->priv->jsessionid = g_strdup(session_id);
     if (proxy->priv->jsessionid != NULL) {
         SoupCookie *cookie;
-        cookie = soup_cookie_new("JSESSIONID", session_id, domain, "/api", -1);
+        cookie = soup_cookie_new("JSESSIONID", session_id, domain, "/ovirt-engine/api", -1);
         soup_cookie_jar_add_cookie(proxy->priv->cookie_jar, cookie);
     }
     g_free(url);
@@ -923,7 +923,7 @@ OvirtApi *ovirt_proxy_fetch_api(OvirtProxy *proxy, GError **error)
 
     g_return_val_if_fail(OVIRT_IS_PROXY(proxy), FALSE);
 
-    api_node = ovirt_proxy_get_collection_xml(proxy, "/api", error);
+    api_node = ovirt_proxy_get_collection_xml(proxy, "/ovirt-engine/api", error);
     if (api_node == NULL) {
         return NULL;
     }
@@ -966,7 +966,7 @@ void ovirt_proxy_fetch_api_async(OvirtProxy *proxy,
     result = g_simple_async_result_new (G_OBJECT(proxy), callback,
                                         user_data,
                                         ovirt_proxy_fetch_api_async);
-    ovirt_proxy_get_collection_xml_async(proxy, "/api", result, cancellable,
+    ovirt_proxy_get_collection_xml_async(proxy, "/ovirt-engine/api", result, cancellable,
                                          fetch_api_async_cb, NULL, NULL);
 }
 
